zebra-utils/src/bin/zebra-checkpoints/main.rs

```diff
@@ -95,12 +95,18 @@ fn main() -> Result<()> {
         .map(BlockHeight)
         .expect("zcashd has some mature blocks: wait for zcashd to sync more blocks");
 
+    let mut starting_height = BlockHeight(args::Args::from_args().last_checkpoint);
+    if starting_height.0 > 0 {
+        starting_height.0 += 1;
+    }
```

Comment:
Suggestion: Let's make the genesis special case part of the type (3/3)

Step 3: Use a mapping to transform:
  * None to BlockHeight(0)
  * Some(BlockHeight(h)) to Some(BlockHeight(h+1))
```suggestion
    // Start at the next block after the last checkpoint.
    // If there is no last checkpoint, start at genesis (height 0).
    let starting_height = starting_height.map_or(0, |BlockHeight(h)| h+1).map(BlockHeight);
```